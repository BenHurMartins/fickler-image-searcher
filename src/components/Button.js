import React from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';
import {Colors, Mixins} from '../styles';

const Button = ({customStyles, onPress, title}) => {
  return (
    <TouchableOpacity
      style={[styles.button, {...customStyles}]}
      onPress={onPress}>
      <Text style={styles.buttonTitle}>{title}</Text>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  button: {
    backgroundColor: Colors.BLACK,
    width: '90%',
    justifyContent: 'center',
    alignItems: 'center',
    height: 50,
    borderRadius: 5,
  },
  buttonTitle: {
    color: Colors.WHITE,
    fontSize: Mixins.FONT_SIZE_REGULAR,
  },
});
