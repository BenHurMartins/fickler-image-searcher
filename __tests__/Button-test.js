import 'react-native';
import React from 'react';
import {render, act} from '@testing-library/react-native';
import Button from '../src/components/Button';
import renderer from 'react-test-renderer';

describe('Button component', () => {
  it('Button renders correctly', () => {
    renderer.create(<Button onPress={() => null} title={'Test'} />);
  });
  it('Button snapshot test', () => {
    const tree = renderer
      .create(<Button onPress={() => null} title={'Test'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should set the title', () => {
    const {getByText} = render(<Button onPress={() => null} title={'Test'} />);
    act(() => {
      expect(getByText('Test')).not.toBe(null);
    });
  });
});
