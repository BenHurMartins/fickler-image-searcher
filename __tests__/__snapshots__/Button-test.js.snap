// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button snapshot test 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": 5,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": "90%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 30,
      }
    }
  >
    Test
  </Text>
</View>
`;
